rubin_scheduler/scheduler/utils/footprints.py

```diff
@@ -589,3 +590,75 @@ def calc_norm_factor_array(goal_map, radius=1.75):
     hp_area = hp.nside2pixarea(nside, degrees=True)
     norm_val = radius**2 * np.pi / hp_area / all_maps_sum
     return norm_val
+
+
+class PerFilterStep(BasePixelEvolution):
+    """Make a custom step function per filter
+    Compensate for filters not always being loaded.
+    This does not use season information, so only
+    useful for a survey of limited duration/limited
+    RA range (e.g., a science validation survey).
+
+    Parameters
+    ----------
+    survey_length : `float`
+      Suervey length in days. Default 80.
+    bands : `list`
+        Bandnames. Defualt of none uses ugrizy
+    loaded_dict : `dict`
+        Dict with keys of bandnames and values of
+        arrays of ints with the nights that band is
+        loaded. E.g.,
+        loaded_dict = {'u': np.array([1,2,3,15,16,17])}
+        means the u band will only be availble on those
+        nights.
```

Comment:
Does this kwarg mean that the times that filters will be loaded over the SV surveys has to be known at the start of the SV surveys, in order for this to work properly?

Reply:
I'm guessing we may not know this exactly (especially for the earlier periods), but maybe getting close enough is fine.

Reply:
Just knowing a filter will be removed for some time in the future should be enough to prevent a big spike in observations.